Fails to build with Clang

Regressed by af54c04bc6a4 (#83)
```c++
../grid/grid_tools.cc:204:22: error: 'dest' in capture list does not name a variable
                    [dest, pos, &view](nop_t) { *dest = view.substr(pos); },
                     ^
../grid/grid_tools.cc:204:28: error: 'pos' in capture list does not name a variable
                    [dest, pos, &view](nop_t) { *dest = view.substr(pos); },
                           ^
../grid/grid_tools.cc:204:50: error: reference to local binding 'dest' declared in enclosing function 'desktop_entry'
                    [dest, pos, &view](nop_t) { *dest = view.substr(pos); },
                                                 ^
../grid/grid_tools.cc:200:29: note: 'dest' declared here
        for (auto& [prefix, dest, tag] : matches) {
                            ^
../grid/grid_tools.cc:204:69: error: reference to local binding 'pos' declared in enclosing function 'desktop_entry'
                    [dest, pos, &view](nop_t) { *dest = view.substr(pos); },
                                                                    ^
../grid/grid_tools.cc:201:23: note: 'pos' declared here
            auto [ok, pos] = try_strip_prefix(prefix);
                      ^
../grid/grid_tools.cc:205:22: error: 'dest' in capture list does not name a variable
                    [dest, pos, &view](cut_t) {
                     ^
../grid/grid_tools.cc:205:28: error: 'pos' in capture list does not name a variable
                    [dest, pos, &view](cut_t) {
                           ^
../grid/grid_tools.cc:206:52: error: reference to local binding 'pos' declared in enclosing function 'desktop_entry'
                        auto idx = view.find(" %", pos);
                                                   ^
../grid/grid_tools.cc:201:23: note: 'pos' declared here
            auto [ok, pos] = try_strip_prefix(prefix);
                      ^
../grid/grid_tools.cc:210:26: error: reference to local binding 'dest' declared in enclosing function 'desktop_entry'
                        *dest = view.substr(pos, idx - pos);
                         ^
../grid/grid_tools.cc:200:29: note: 'dest' declared here
        for (auto& [prefix, dest, tag] : matches) {
                            ^
../grid/grid_tools.cc:210:45: error: reference to local binding 'pos' declared in enclosing function 'desktop_entry'
                        *dest = view.substr(pos, idx - pos);
                                            ^
../grid/grid_tools.cc:201:23: note: 'pos' declared here
            auto [ok, pos] = try_strip_prefix(prefix);
                      ^
../grid/grid_tools.cc:210:56: error: reference to local binding 'pos' declared in enclosing function 'desktop_entry'
                        *dest = view.substr(pos, idx - pos);
                                                       ^
../grid/grid_tools.cc:201:23: note: 'pos' declared here
            auto [ok, pos] = try_strip_prefix(prefix);
                      ^
10 errors generated.
```

--- a/grid/grid_tools.cc
+++ b/grid/grid_tools.cc
@@ -156,8 +156,8 @@ std::optional<DesktopEntry> desktop_entry(std::string&& path, const std::string&
         std::variant<nop_t, cut_t> tag;
     };
     struct Result {
-        bool   found;
-        size_t index;
+        bool   ok;
+        size_t pos;
     };
     Match matches[] = {
         { "Name="sv,     &entry.name,      nop },
@@ -197,9 +197,11 @@ std::optional<DesktopEntry> desktop_entry(std::string&& path, const std::string&
             // See https://wiki.archlinux.org/index.php/desktop_entries#Hide_desktop_entries
             entry.no_display = true;
         }
-        for (auto& [prefix, dest, tag] : matches) {
-            auto [ok, pos] = try_strip_prefix(prefix);
-            if (ok) {
+        for (auto& match : matches) {
+            auto result = try_strip_prefix(match.prefix);
+            auto& dest = match.dest; // it was 2020, clang failed to capture
+            auto  pos  = result.pos; // var from structured binding, so we had to write it by hand
+            if (result.ok) {
                 std::visit(visitor {
                     [dest, pos, &view](nop_t) { *dest = view.substr(pos); },
                     [dest, pos, &view](cut_t) {
@@ -210,7 +212,7 @@ std::optional<DesktopEntry> desktop_entry(std::string&& path, const std::string&
                         *dest = view.substr(pos, idx - pos);
                     }
                 },
-                tag);
+                match.tag);
                 break;
             }
         }

